### ISSUE TITLE
Type is not hashable error, when distribution key is an array

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#3741]
```
postgres=# create table text_array_table (t text[]);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 't' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE TABLE
postgres=# insert into text_array_table values ('{foo}');
ERROR:  Type 1009 is not hashable.  (seg2 127.0.0.1:40002 pid=23033)
```

**Comment by vraghavan78** [comment]
Thanks for reporting. This issue existed in 4.3 too.

**Comment by hlinnaka** [comment]
This was fixed by PR #4940, closing.
